### ISSUE TITLE
Image stuck on first item

### ISSUE THREAD

**Issue by jeffski13** [xiaolin/react-image-gallery#433]
I updated to 0.8.18 and the gallery keeps going back to first image. Pressing play has the same results when it tried to go to the next image.

Here is how i am using it:
```
import ImageGallery from 'react-image-gallery'
import "react-image-gallery/styles/css/image-gallery.css";

<ImageGallery 
    thumbnailPosition="bottom" 
    items={imageUrls} 
    onSlide={(index) => {
                        this.setState({
                            currentImageIndex: index
                        });
    }} 
/>
```

image urls are and array of 
```
{
    original: "string",
    thumbnail: "string",
    originalTitle: "string",
    thumbnailTitle: "string"
}
```

**Comment by xiaolin** [comment]
The demo page is using version 0.8.18 and I'm unable to replicate. Can you remove your `onSlide` function and see if its still happening?

**Comment by nganthanhnguyen** [comment]
I'm faced with the same issue. Update state inside onSlide event will cause this issue

**Comment by tung2389** [comment]
In the build/image-gallery.js, I comment out this code in `  _createClass` and the image no longer stuck in the first item

`if (startIndexUpdated || itemsChanged) {
        this.setState({ currentIndex: this.props.startIndex });
}`

**Comment by xiaolin** [comment]
This bug has been fixed in `v0.9.1`. Additionally, this bug can be avoided if you guys move `items` outside of the render method.

```
render () {
  const items = [...];
  return <ImageGallery items={items}/>
}

to

constructor() {
   super();
   this.items = [...];
}

render () {
  return <ImageGallery items={this.items}
}
```
